### DISCUSSION TITLE
Thebe like feature

### DISCUSSION THREAD

**Discussion by fbob** [quarto-dev/quarto-cli#5573] [Feature Requests]
Sorry if this feature is already available
I wonder if a [thebe](https://github.com/executablebooks/thebe) like functionality could be implemented for Quarto ?

For those who don't know what thebe is (taken from the official doc) :

>Thebe is a set of libraries allowing web applications and static web pages to provide interactive computation backed by a Juptyer kernel.

**Comment by mcanouil** [comment]
As for WebR Quarto extension (<https://github.com/coatless/quarto-webr>) for WebR (WASM), these can be embedded into a Quarto extension, see <https://quarto.org/docs/extensions/creating.html>.
I personally don't think this should be integrated directly in Quarto, thus maintain by Quarto team.

**Reply by fbob** [reply]
Thank you, @mcanouil, for this helpful feedback. I didn't know about this WebR extension.
Any idea if a Python version could also be technically implemented as an extension for Quarto ? 

Anyway, many thanks for your feedback :+1:

**Reply by mcanouil** [reply]
Every HTML thing should work as a Quarto extension.

Quarto extension is simply a mechanism to help everybody get the best of what HTML (or other format) has to offer.

**Comment by psychemedia** [comment]
There is a simple quarto extension integrating pyodide into HTML pages to support in-page code execution at https://github.com/coatless-quarto/pyodide .

The organisation currently also has an, as yet empty, placeholder repo for what looks like a possible thebe extension for quarto.

**Reply by fbob** [reply]
@psychemedia many thanks for the info
